jib-gradle-plugin/src/main/java/com/google/cloud/tools/jib/gradle/TargetImageParameters.java

```diff
@@ -69,12 +69,21 @@ public void setImage(Provider<String> image) {
   @Input
   @Optional
   public Set<String> getTags() {
-    if (System.getProperty(PropertyNames.TO_TAGS) != null) {
-      return ImmutableSet.copyOf(
-          ConfigurationPropertyValidator.parseListProperty(
-              System.getProperty(PropertyNames.TO_TAGS)));
+    String property = System.getProperty(PropertyNames.TO_TAGS);
+    Set<String> tagsValue;
+    if (property != null) {
+      tagsValue = ImmutableSet.copyOf(ConfigurationPropertyValidator.parseListProperty(property));
+    } else {
+      try {
+        tagsValue = tags.get();
+      } catch (NullPointerException ex) {
+        throw new IllegalArgumentException("jib.to.tags contains null tag");
+      }
     }
-    return tags.get();
+    if (tagsValue.stream().anyMatch(str -> str.isEmpty())) {
+      throw new IllegalArgumentException("jib.to.tags contains empty tag");
```

Comment:
Do we have to do that logic where we know its from system property or from configuration in teh build file?

Reply:
In terms of error messaging, I didn't think having that logic added much value since regardless of whether it was from system property or configuration, (in `String source = property != null ? PropertyNames.TO_TAGS : "jib.to.tags";`) the message would still remain the same?